src/components/_global/BalChip/BalChip.vue

```diff
@@ -1,132 +1,61 @@
+<script setup lang="ts">
+import { withDefaults, defineEmits } from 'vue';
+
+import useChipClasses from './composables/useChipClasses';
+import useCloseIconClasses from './composables/useCloseIconClasses';
+
+import BalIcon from '../BalIcon/BalIcon.vue';
+
+export type Props = {
+  label?: string;
+  closeable?: boolean;
+  size?: 'sm' | 'md' | 'lg';
+  color?: 'gray' | 'gradient' | 'white' | 'red';
+  outline?: boolean;
+  rounded?: boolean;
+};
+
+/**
+ * PROPS & EMITS
+ */
+const props = withDefaults(defineProps<Props>(), {
+  label: '',
+  closeable: false,
+  size: 'md',
+  color: 'gray',
+  rounded: false,
+  outline: true
+});
+
+const emit = defineEmits<{
+  (e: 'closed', value: any): void;
```

Comment:
@garethfuller the `value` here is the `@click` event - any ideas where I find the type for it?